command/operator_debug.go

```diff
@@ -460,51 +488,298 @@ func (c *OperatorDebugCommand) Run(args []string) int {
 	return 0
 }
 
-// collect collects data from our endpoints and writes the archive bundle
-func (c *OperatorDebugCommand) collect(client *api.Client) error {
-	// Version contains cluster meta information
-	dir := "version"
+// targetMap generic map for targets
+type targetMap map[string]bool
 
-	self, err := client.Agent().Self()
-	c.writeJSON(dir, "agent-self.json", self, err)
+// String returns a CSV of valid enabled targets
+func (t targetMap) String() string {
+	var targets []string
+	var targetString string
+
+	for target, enabled := range t {
+		if enabled && target != "all" {
+			targets = append(targets, target)
+		}
+	}
+	sort.Strings(targets)
+	targetString = strings.Join(targets, ",")
+
+	return targetString
+}
+
+// IsValid checks whether the key exists in the map
+func (m targetMap) IsValid(value string) bool {
+	if value == "all" {
+		return true
+	}
+	_, ok := m[value]
+	return ok
+}
+
+// parseTargets parses a comma delimited target string into a targetMap
+func (c *OperatorDebugCommand) parseTargets(targets string) targetMap {
+	ret := make(targetMap)
+
+	// Build starting target map
+	switch {
+	case targets == "":
+		return ret
+	case strings.Contains(targets, "none"):
+		ret = targetDefaultsNone()
```

Comment:
What should happen when a user passed `none, allocations`? Should we generate an error? Should we just return `targetDefaultsNone()` here as well without further processing?

Reply:
As a draft PR I hadn't fully explained the concept yet, but the idea here is that the target flags are additive / subtractive.  If you pass no target flag at all, then the default behavior is the same as it was before and "all" targets are enabled.  To clarify I will add a description to the PR and expand the documentation in a future commit.

To your question, `none,allocations` should result in a map containing only `allocations`.  To verify and help self-document I will add a test to ensure that this is always the case.

As another example, I have an upcoming PR which adds an eventstream target that will be disabled by default.  To add this target to a capture you would use `-target=eventstream`.  This results in a map containing all target defaults plus eventstream.  Similar to your question, if you wanted _only_ eventstream then you would use `-target=none,eventstream`.